### PR TITLE
Various fixes and feature updates for V2 Reconfigure

### PR DESCRIPTION
**Pull request by wraeth-eth**
## What does this PR do and why?

Closes https://github.com/jbx-protocol/juice-interface/issues/888
Closes https://github.com/jbx-protocol/juice-interface/issues/930
Closes https://github.com/jbx-protocol/juice-interface/issues/917

Adds modal feedback for unsaved changes to reconfigure drawers and reconfigure modal.

### Modal when you first hit reconfigure
Screen Shot 2022-05-06 at 3 23 31 pm

### Modal after making a change in a drawer
Screen Shot 2022-05-06 at 3 23 42 pm

### Drawer Unsaved Changes Modal
Screen Shot 2022-05-06 at 3 23 52 pm

### Global Modal Unsaved Changes Modal
Screen Shot 2022-05-06 at 3 24 08 pm

### Video Demo
https://user-images.githubusercontent.com/104132113/167073383-7b7ed975-d5e8-4c11-ba01-9203365344e0.mov

## Acceptance checklist

- [x] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [x] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).

### CODE REVIEW

File: src/components/v2/V2Project/V2ProjectReconfigureModal/index.tsx [hunk 1]
```diff
@@ -158,8 +227,10 @@ export default function V2ProjectReconfigureModal({
     ? queuedReservedTokensSplits
     : reservedTokensSplits
 
-  const effectiveDistributionLimit =
-    queuedDistributionLimit ?? distributionLimit
+  let effectiveDistributionLimit = distributionLimit
+  if (effectiveFundingCycle?.duration.gt(0) && queuedDistributionLimit) {
```
Comment (wraeth-eth) [inline src/components/v2/V2Project/V2ProjectReconfigureModal/index.tsx:231]: Fixes: https://github.com/jbx-protocol/juice-interface/issues/930

File: src/components/v2/V2Project/V2ProjectReconfigureModal/index.tsx [hunk 2]
```diff
@@ -272,32 +395,33 @@ export default function V2ProjectReconfigureModal({
         },
         onConfirmed() {
           setReconfigureTxLoading(false)
-          onOk()
+          exit()
         },
       },
     )
+
+    if (!txSuccessful) {
```
Comment (wraeth-eth) [inline src/components/v2/V2Project/V2ProjectReconfigureModal/index.tsx:403]: Fixes https://github.com/jbx-protocol/juice-interface/issues/917

File: src/components/v2/V2Project/V2ProjectReconfigureModal/index.tsx (hunk 1 above)
Comment (johnnyd-eth) [reply src/components/v2/V2Project/V2ProjectReconfigureModal/index.tsx:231]: Ah yep nice 🙌

File: src/components/v2/V2Project/V2ProjectReconfigureModal/index.tsx (hunk 2 above)
Comment (johnnyd-eth) [reply src/components/v2/V2Project/V2ProjectReconfigureModal/index.tsx:403]: Yes very nice, been hit with this a few times

File: src/components/v2/V2Project/V2ProjectReconfigureModal/index.tsx [hunk 3]
```diff
@@ -272,32 +395,33 @@ export default function V2ProjectReconfigureModal({
...
     editingFundingCycleData,
     reconfigureV2FundingCycleTx,
     editingPayoutGroupedSplits,
     editingReservedTokensGroupedSplits,
-    onOk,
+    exit,
   ])
 
   const saveFundingTab = () => {
-    setFundingHasSavedChanges(true)
```
Comment (johnnyd-eth) [inline src/components/v2/V2Project/V2ProjectReconfigureModal/index.tsx:290]: **Question**: why are we getting rid of these? Are they not needed for your recent 'you have unsaved changes' PR?

**Review by johnnyd-eth — APPROVED** [review]
Beautiful @wraeth-eth all perfect fixes to the mentioned problems, legend. 🚀

**Pull request merged**